scripts/tess/build_parent_sample.py

```diff
@@ -0,0 +1,196 @@
+        os.makedirs(os.path.dirname(output_filename))
+
+    # Rename columns to match the standard format
+    catalog['object_id'] = catalog['target_name']
+    
+    # Extract the light curves by looping over all files
+    # catalog = catalog.group_by(['sector'])
+    
+    # Preparing the arguments for the parallel processing
+    # map_args = []
+    #for group in catalog.groups:
+        #pipeline = group['pipeline'][0]
+        #sector = group['sector'][0]
+        #object_id = group['object_id']
+        #file_path = group['lc_path']
+        #map_args += [(file_path, object_id)]
+
+    # Process all files
+
+    results = []
+    for args in catalog[['lc_path', 'object_id']]:
+        results.append(processing_fn(args))
+
+    
+    # Pad all light curves to the same length
+    max_length = max([len(d['time']) for d in results])
+    for i in range(len(results)):
+        results[i]['time'] = np.pad(results[i]['time'], (0,max_length - len(results[i]['time'])), mode='constant')
+        results[i]['flux'] = np.pad(results[i]['flux'], (0,max_length - len(results[i]['flux'])), mode='constant')
+        results[i]['flux_err'] = np.pad(results[i]['flux_err'], (0,max_length - len(results[i]['flux_err'])), mode='constant')
+    
+    # Aggregate all light curves into an astropy table
+    lightcurves = Table({k: [d[k] for d in results]
+                     for k in results[0].keys()})
+
+    # Join on target id with the input catalog
+    catalog = join(catalog, lightcurves, keys='object_id', join_type='inner')
+    
+    # Making sure we didn't lose anyone
+    assert len(catalog) == len(lightcurves), "There was an error in the join operation, probably some spectra files are missing"
+
+    #TODO: Solve error underneath
+    # Save all columns to disk in HDF5 format
+    with h5py.File(output_filename, 'w') as hdf5_file:
+        for key in catalog.colnames:
+            hdf5_file.create_dataset(key, data=catalog[key])
+    return 1
+
+def main(args):
+    # Load the catalog file and apply main cuts
+    catalog = Table.read(os.path.join(args.tess_data_path, "tess_lc_catalog_sector_64.csv"))
+    
+    #TODO: catalog = catalog[selection_fn(catalog)]
+
+    # Add healpix index to the catalog
+    catalog['healpix'] = hp.ang2pix(64, catalog['RA'], catalog['DEC'], lonlat=True, nest=True)
+
+    #TODO: add support for multiple pipelines, currently only using SPOC
+    for pipeline in PIPELINES:
+        print("Processing pipeline:", pipeline)
+
+        cat_pipeline = catalog
+        #cat_pipeline = catalog[catalog['PIPELINE'] == pipeline]
+        
+        cat_pipeline = cat_pipeline.group_by(['healpix'])
+    
+        map_args = []
+        for group in cat_pipeline.groups:
+            # Create a filename for the group
+            group_filename = os.path.join(args.output_dir, '{}/healpix={}/001-of-001.hdf5'.format(pipeline.strip(), group['healpix'][0]))
+            if args.tiny:
```

Comment:
don't need this if statement, args.tiny will resolve to false if the --tiny flag is not set

Reply:
Solved, thanks!